plugins/bigquery/dbt/adapters/bigquery/connections.py

```diff
@@ -305,12 +307,24 @@ def raw_execute(self, sql, fetch=False, *, use_legacy_sql=False):
 
         logger.debug('On {}: {}', conn.name, sql)
 
-        job_params = {'use_legacy_sql': use_legacy_sql}
+        labels = {}
 
         if active_user:
-            job_params['labels'] = {
-                'dbt_invocation_id': active_user.invocation_id
-            }
+            labels['dbt_invocation_id'] = active_user.invocation_id
+
+        if self.profile.query_comment.job_label:
+            try:
+                comment_labels = json.loads(
+                    self.query_header.comment.query_comment
+                )
+                labels.update({
+                    _sanitize_label(key): _sanitize_label(str(value))
+                    for key, value in comment_labels.items()
+                })
+            except (TypeError, ValueError):
+                pass
```

Comment:
If the comment isn't valid json, should we still include it as a single job label (called `query_comment`), truncated to 128 characters if need be?